mava/components/jax/building/reverb.py

```diff
@@ -0,0 +1,243 @@
+        """_summary_
+
+        Args:
+            builder : _description_
+        """
+
+        def rate_limiter_fn() -> reverb.rate_limiters:
+            return reverb.rate_limiters.MinSize(self.config.min_data_server_size)
+
+        builder.store.rate_limiter_fn = rate_limiter_fn
+
+
+class SampleToInsertRateLimiter(RateLimiter):
+    def on_building_data_server_rate_limiter(self, builder: SystemBuilder) -> None:
+        """_summary_
+
+        Args:
+            builder : _description_
+        Returns:
+            _description_
+        """
+        if not self.config.error_buffer:
+            # Create enough of an error buffer to give a 10% tolerance in rate.
+            samples_per_insert_tolerance = 0.1 * self.config.samples_per_insert
+            error_buffer = (
+                self.config.min_data_server_size * samples_per_insert_tolerance
+            )
+        else:
+            error_buffer = self.config.error_buffer
+
+        def rate_limiter_fn() -> reverb.rate_limiters:
+            return reverb.rate_limiters.SampleToInsertRatio(
+                min_size_to_sample=self.config.min_data_server_size,
+                samples_per_insert=self.config.samples_per_insert,
+                error_buffer=error_buffer,
+            )
+
+        builder.store.rate_limiter_fn = rate_limiter_fn
+
+
+@dataclass
+class SamplerConfig:
+    pass
+
+
+class Sampler(Component):
+    def __init__(
+        self,
+        config: SamplerConfig = SamplerConfig(),
+    ):
+        """_summary_
+
+        Args:
+            config : _description_.
+        """
+        self.config = config
+
+    @abc.abstractmethod
+    def on_building_data_server_start(self, builder: SystemBuilder) -> None:
+        """_summary_"""
+
+    @staticmethod
+    def name() -> str:
+        """_summary_
+
+        Returns:
+            _description_
+        """
+        return "data_server_sampler"
+
+    @staticmethod
+    def config_class() -> Optional[Callable]:
+        """Config class used for component.
+
+        Returns:
+            config class/dataclass for component.
+        """
+        return SamplerConfig
+
+
+class UniformSampler(Sampler):
+    def on_building_data_server_start(self, builder: SystemBuilder) -> None:
+        """_summary_"""
+
+        def sampler_fn() -> reverb.selectors:
+            return reverb.selectors.Uniform()
+
+        builder.store.sampler_fn = sampler_fn
+
+
+@dataclass
+class PrioritySamplerConfig:
+    priority_exponent: float = 1.0
+
+
+class PrioritySampler(Sampler):
+    def __init__(
+        self,
+        config: PrioritySamplerConfig = PrioritySamplerConfig(),
+    ):
+        """_summary_
+
+        Args:
+            config : _description_.
+        """
+        self.config = config
+
+    def on_building_data_server_start(self, builder: SystemBuilder) -> None:
+        """_summary_"""
+        builder.store.priority_exponent = self.config.priority_exponent
```

Comment:
This can now be handled through `store.global_config` instead. Please see elsewhere in the code.